cpp/src/parquet/encoding_test.cc

```diff
@@ -739,7 +739,8 @@ std::shared_ptr<arrow::Array> EncodingAdHocTyped<FLBAType>::GetValues(int seed)
   std::shared_ptr<arrow::Array> values;
   ARROW_EXPECT_OK(
```

Comment:
Why not use `ASSERT_OK_AND_ASSIGN(auto values, ...)`?

Reply:
Because ASSERT macros only work in `void`-returning function.

Reply:
FWIW, we also have EXPECT_OK_AND_ASSIGN

Reply:
Ah, I missed that!

Reply:
I don't like those macros, though, because they can crash the rest of the test (if the OK part fails, the LHS will be left unassigned, though it's probably used later).